src/tests/test_rest_framework_dso/test_serializers.py

```diff
@@ -58,6 +59,31 @@ def test_serializer_many(api_request, movie):
     }
 
 
+@pytest.mark.django_db
+def test_serializer_embed_with_missing_relations(api_request):
+    """Prove that the serializer can embed data (for the detail page)"""
+
+    cursor = connection.cursor()
+    cursor.execute(
+        "INSERT INTO test_rest_framework_dso_movie (name, category_id) VALUES ('Test', 333);"
+    )
+    movie = Movie.objects.get(name="Test")
+
+    serializer = MovieSerializer(
+        many=True,
+        instance=[movie],
+        fields_to_expand=["category"],
+        context={"request": api_request},
+    )
+    assert serializer.data == {
+        "movie": [{"name": "Test", "category_id": movie.category_id}],
+        "category": [],
+    }
+
+    # Cleanup needed to make Django happy.
+    cursor.execute("DELETE FROM test_rest_framework_dso_movie")
```

Comment:
Dit is vreemd toch? Je zou verwachten dat tests voldoende geisoleerd zijn en via teardown dit regelen.

Reply:
Maar het gaat in teardown mis :) Django probeerd CASCADE te doen, waar ook relaties gecheked zijn, en door missende relatie breekt de test :)

```
    def _post_teardown(self):
        """
        Perform post-test things:
        * Flush the contents of the database to leave a clean slate. If the
          class has an 'available_apps' attribute, don't fire post_migrate.
        * Force-close the connection so the next test gets a clean cursor.
        """
        try:
>           self._fixture_teardown()
....

                # params default might be backend specific.
>               return self.cursor.execute(sql)
E               django.db.utils.IntegrityError: insert or update on table "test_rest_framework_dso_movie" violates foreign key constraint "test_rest_framework__category_id_4cf82834_fk_test_rest"
E               DETAIL:  Key (category_id)=(333) is not present in table "test_rest_framework_dso_category".

../../../../.virtualenvs/dso-api/lib/python3.8/site-packages/django/db/backends/utils.py:84: IntegrityError
```

Reply:
Ah vandaar, een beetje een bijzonder geval.